runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -139,6 +160,29 @@ private static void setupSlf4jReporting(MetricRegistry metrics) {
         slf4jReporter.start(metricsLogInterval, TimeUnit.SECONDS);
     }
 
+    // If enabled, setup csv reporting
+    private static void setupCsvReporting(MetricRegistry metrics) {
+        if (!metricsCsvReportingEnabled) return;
+
+        final File directory = new File(metricsCsvFolder);
+
+        if (!directory.isDirectory() ||
+            !directory.exists() ||
+            !directory.canWrite()) {
+            log.warn("Provided CSV directory : {} doesn't exist, isn't a directory, or not accessible " +
+                    "for writes. Disabling CSV Reporting.");
+            metricsCsvReportingEnabled = false;
```

Comment:
Why fail silently here? if this is enabled then its expected to work no ?

Reply:
Logging the failure to collect CSV metrics due to non existing directory or lack of write privilege seems to be reasonable response to the failure. It is probably excessive to consider it as a show stopper.